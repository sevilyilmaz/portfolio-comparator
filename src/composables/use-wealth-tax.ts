// https://www.belastingdienst.nl/wps/wcm/connect/nl/box-3/content/nieuwe-berekening-box-3-inkomen
const wealthTax = 36 / 100;
const taxFreeWealth = 57000;
const fictitiusReturnPercentage = 6.04 / 100;

export function getWealthTax(amount: number): number {
  if (amount <= taxFreeWealth) return 0;
  const fictitiusReturn = amount * fictitiusReturnPercentage;
  const taxableAmount = amount - taxFreeWealth;
  const taxableContributionPercentage = taxableAmount / amount;
  const fictitiusGain = fictitiusReturn * taxableContributionPercentage;

  return fictitiusGain * wealthTax;
}

export function substractWealthTax(
  amount: number,
  fiscalStatus: number
): number {
  const amountPerPerson = amount / fiscalStatus;
  const tax = getWealthTax(amountPerPerson) * fiscalStatus;
  return amount - tax;
}
