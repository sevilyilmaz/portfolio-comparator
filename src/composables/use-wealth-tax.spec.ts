import { describe, test, expect } from 'vitest';
import { getWealthTax, substractWealthTax } from './use-wealth-tax';

describe('getWealthTax', () => {
  test.each([
    28000, 50000, 80000, 100000, 120000, 150000, 950000, 1200000, 200000,
  ])('should return the wealth tax for [%s]', (amount) => {
    expect(getWealthTax(amount)).toMatchSnapshot();
  });
});

describe('substractWealthTax', () => {
  test.each([28000, 50000, 80000, 100000, 120000, 950000, 1200000])(
    'should substract wealth tax from [%s] for single person',
    (amount) => {
      expect(substractWealthTax(amount, 1)).toMatchSnapshot();
    }
  );

  test.each([28000, 50000, 80000, 100000, 120000, 950000, 1200000])(
    'should substract wealth tax from [%s] for a couple (50%/50% split)',
    (amount) => {
      expect(substractWealthTax(amount, 2)).toMatchSnapshot();
    }
  );
});
