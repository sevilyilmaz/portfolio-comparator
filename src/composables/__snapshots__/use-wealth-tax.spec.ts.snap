// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getWealthTax > should return the wealth tax for [28000] 1`] = `0`;

exports[`getWealthTax > should return the wealth tax for [50000] 1`] = `0`;

exports[`getWealthTax > should return the wealth tax for [80000] 1`] = `165.41073`;

exports[`getWealthTax > should return the wealth tax for [100000] 1`] = `278.12673`;

exports[`getWealthTax > should return the wealth tax for [120000] 1`] = `538.5502899999999`;

exports[`getWealthTax > should return the wealth tax for [950000] 1`] = `11772.26829`;

exports[`getWealthTax > should return the wealth tax for [1200000] 1`] = `16013.45455`;

exports[`substractWealthTax > should substract wealth tax from [28000] for a couple (50%/50% split) 1`] = `28000`;

exports[`substractWealthTax > should substract wealth tax from [28000] for single person 1`] = `28000`;

exports[`substractWealthTax > should substract wealth tax from [50000] for a couple (50%/50% split) 1`] = `50000`;

exports[`substractWealthTax > should substract wealth tax from [50000] for single person 1`] = `50000`;

exports[`substractWealthTax > should substract wealth tax from [80000] for a couple (50%/50% split) 1`] = `80000`;

exports[`substractWealthTax > should substract wealth tax from [80000] for single person 1`] = `79834.58927`;

exports[`substractWealthTax > should substract wealth tax from [100000] for a couple (50%/50% split) 1`] = `100000`;

exports[`substractWealthTax > should substract wealth tax from [100000] for single person 1`] = `99721.87327`;

exports[`substractWealthTax > should substract wealth tax from [120000] for a couple (50%/50% split) 1`] = `119894.61054`;

exports[`substractWealthTax > should substract wealth tax from [120000] for single person 1`] = `119461.44971`;

exports[`substractWealthTax > should substract wealth tax from [950000] for a couple (50%/50% split) 1`] = `939313.33342`;

exports[`substractWealthTax > should substract wealth tax from [950000] for single person 1`] = `938227.73171`;

exports[`substractWealthTax > should substract wealth tax from [1200000] for a couple (50%/50% split) 1`] = `1185929.68342`;

exports[`substractWealthTax > should substract wealth tax from [1200000] for single person 1`] = `1183986.54545`;
